react-native-git-upgrade/cliEntry.js

```diff
@@ -254,9 +265,19 @@ async function run(requestedVersion, cliArgs) {
     log.info('Commit old version template');
     await exec('git commit -m "Old version" --allow-empty', verbose);
 
-    log.info('Install the new version');
+    oldReactPeerDepVersion = getReactPeerDependencyVersion();
+
+    log.info('Install the new version: react-native@' + newVersion);
     await exec('npm install --save react-native@' + newVersion + ' --color=always', verbose);
 
+    const newReactPeerDepVersion = getReactPeerDependencyVersion();
+    if (newReactPeerDepVersion) {
+      log.info('Install the corresponding React version (peer dependency): react@' + newReactPeerDepVersion);
+      await exec('npm install --save react@' + newReactPeerDepVersion + ' --color=always', verbose);
```

Comment:
If React is upgraded after RN, NPM2 will still fail in case of unmet `react` peerDep during the previous step. I think the only way to fix that is to upgrade both R & RN at the same time. See: https://github.com/facebook/react-native/pull/11123#issuecomment-262850815